### PR TITLE
Add listening post access to sleepers

### PR DESCRIPTION
**Pull request by cheekybrdy**
## About the PR 
This is a rework of https://github.com/goonstation/goonstation/pull/15222 which is a rework of https://github.com/goonstation/goonstation/pull/14992 because I messed something up again. This PR adds handscanners to the listening post which allow sleeper agents to leave and enter it.

## Why's this needed? 
This gives sleepers access to items in the listening post giving them traitor items without the full power of the traitor uplink. This makes them feel like more then a late HMT.

## Changelog 

```changelog
(u)Cheekybrdy
(*)Sleeper agents can now access the listening post through scanners at the entrance.
```

### CODE REVIEW

**Comment by cheekybrdy** [comment]
Discussion thread: https://forum.ss13.co/showthread.php?tid=21546

File: code/obj/machinery/sleeperscanner.dm [hunk 1]
```diff
@@ -0,0 +1,25 @@
+/obj/machinery/sleeperscanner
+	name = "Hand Scanner"
+	icon = 'icons/obj/decoration.dmi'
+	icon_state = "handscanner"
+	desc = "A device embedded into the wall, it has a image of a handprint on it."
+
+/obj/machinery/sleeperscanner/attackby(obj/item/W, mob/user as mob)
+	if(istype(W, /obj/item/device/detective_scanner))
+		return
+	if (user.mind?.get_antagonist(ROLE_SLEEPER_AGENT) && user.mind?.get_antagonist(ROLE_SLEEPER_AGENT))
```
Comment (Mister-Moriarty) [inline code/obj/machinery/sleeperscanner.dm:10]:
```suggestion
	if (ishuman(user) && user.mind?.get_antagonist(ROLE_SLEEPER_AGENT))
```

**Review by Mister-Moriarty — CHANGES_REQUESTED** [review]
I would again recommend refactoring `/obj/machinery/sleeperscanner` into `/obj/machinery/door_control/sleeperscanner`, and matching the `id` var on both the sleeper scanner and on the `/obj/machinery/door/airlock/pyro/reinforced/syndicate/listening_post` type to a unique string, alongside the rather minor required code changes.

This refactor would avoid the duplication of code, as your sleeper hand scanners operate in a similar manner to door controls.

**Review by Mister-Moriarty — CHANGES_REQUESTED** [review]
You appear to have removed several files from `goonstation.dme`, which is likely causing the runtimes.

If you have not already, I would also recommend installing the `.dmm` and `.dmi` merge drivers, as you appear to be having issues with merging upstream map changes.

**Comment by cheekybrdy** [comment]
This lgtm

File: code/obj/machinery/door/door_control.dm [hunk 2]
```diff
@@ -479,6 +486,42 @@ ADMIN_INTERACT_PROCS(/obj/machinery/door_control, proc/toggle)
...
+	icon_state = "antagscanner"
+	unpressed_icon = "antagscanner"
+	pressed_icon = "antagscanner-u"
+	unpowered_icon = "antagscanner"
+	requires_power = 0
+	welcome_text = "Welcome, Agent."
+
+/obj/machinery/door_control/antagscanner/attack_hand(mob/user)
+	playsound(src.loc, 'sound/effects/handscan.ogg', 50, 1)
+	if (ishuman(user) && user.mind?.get_antagonist(ROLE_SLEEPER_AGENT))
```
Comment (TobleroneSwordfish) [inline code/obj/machinery/door/door_control.dm:517]:
```suggestion
	if (user.mind?.get_antagonist(ROLE_SLEEPER_AGENT))
```
no particular reason to have this, may as well allow goose sleeper agents

File: code/obj/machinery/door/door_control.dm [hunk 3]
```diff
@@ -479,6 +486,42 @@ ADMIN_INTERACT_PROCS(/obj/machinery/door_control, proc/toggle)
 	else
 		boutput(user, "<span class='alert'>It's broken.</span>")
 
+/obj/machinery/door_control/proc/speak(var/message) //Stolen from the vending module
+	var/image/chat_maptext/welcome_text
```
Comment (TobleroneSwordfish) [inline code/obj/machinery/door/door_control.dm:490]: why is this defined on the type and as a local?

File: code/obj/machinery/door/door_control.dm [hunk 4]
```diff
@@ -479,6 +482,55 @@ ADMIN_INTERACT_PROCS(/obj/machinery/door_control, proc/toggle)
...
+
+	if (!text_out)
+		return
+	else
+		src.audible_message("<span class='subtle'><span class='game say'><span class='name'>[src]</span> beeps, \"[text_out]\"</span></span>", 2, assoc_maptext = slogan_text)
+
+	return
+
+/obj/machinery/door_control/antagscanner // for sleepers entering listening post
+	var/entrance_scanner = 0 // Do we want a welcome message or similar for the area upon activation?
```
[the remark this answers is not in the document]
Comment (TobleroneSwordfish) [reply code/obj/machinery/door/door_control.dm:504]: for reference, doc comments look like this
```
///This is a doc comment
var/foo = 3
```

**Review by TobleroneSwordfish — CHANGES_REQUESTED** [review]

**Review by TobleroneSwordfish — CHANGES_REQUESTED** [review]
- You've accidentally deleted some walls on atlas and cog 1, see the mapdiffbot renders in `Checks` for details.
- The scanner seems to only speak the first time it's used, and says "" every time afterwards
- It can be spammed a bit, it's good practice to put an `ON_COOLDOWN` check on anything that generates a chat message so it doesn't flood the chat box

File: code/obj/machinery/door/door_control.dm [hunk 5]
```diff
@@ -478,6 +490,47 @@ ADMIN_INTERACT_PROCS(/obj/machinery/door_control, proc/toggle)
...
+		src.audible_message("<span class='subtle'><span class='game say'><span class='name'>[src]</span> beeps, \"[message]\"</span></span>", 2, assoc_maptext = welcome_text)
+		if (welcome_text && src.chat_text && length(src.chat_text.lines))
+			welcome_text.measure(src)
+			ON_COOLDOWN(src, "scanner_speak", 2)
+			for (var/image/chat_maptext/I in src.chat_text.lines)
+				if (I != welcome_text)
+					I.bump_up(welcome_text.measured_height)
+					ON_COOLDOWN(src, "scanner_speak", 2)
+		ON_COOLDOWN(src, "scanner_speak", 2)
+	return
```
Comment (TobleroneSwordfish) [inline code/obj/machinery/door/door_control.dm:510]:
```suggestion
```

File: code/obj/machinery/door/door_control.dm [hunk 6]
```diff
@@ -478,6 +490,47 @@ ADMIN_INTERACT_PROCS(/obj/machinery/door_control, proc/toggle)
...
+/obj/machinery/door_control/proc/speak(var/message)
+	ON_COOLDOWN(src, "scanner_speak", 2)
+	if ((src.status & NOPOWER) || !message)
+		return
+	else
+		welcome_text = make_chat_maptext(src, message, "color: [src.welcome_text_color];", alpha = src.welcome_text_alpha)
+		src.audible_message("<span class='subtle'><span class='game say'><span class='name'>[src]</span> beeps, \"[message]\"</span></span>", 2, assoc_maptext = welcome_text)
+		if (welcome_text && src.chat_text && length(src.chat_text.lines))
+			welcome_text.measure(src)
+			ON_COOLDOWN(src, "scanner_speak", 2)
```
Comment (TobleroneSwordfish) [inline code/obj/machinery/door/door_control.dm:504]:
```suggestion
```

File: code/obj/machinery/door/door_control.dm [hunk 7]
```diff
@@ -478,6 +490,47 @@ ADMIN_INTERACT_PROCS(/obj/machinery/door_control, proc/toggle)
 		playsound(src.loc, 'sound/impact_sounds/Generic_Click_1.ogg', 50, 1)
 	else
 		boutput(user, "<span class='alert'>It's broken.</span>")
+// Stolen from the vending module
+/// For a flying chat and message addition upon controller activation, not called outside of a child as things stand
+/obj/machinery/door_control/proc/speak(var/message)
+	ON_COOLDOWN(src, "scanner_speak", 2)
```
Comment (TobleroneSwordfish) [inline code/obj/machinery/door/door_control.dm:496]:
```suggestion
```

File: code/obj/machinery/door/door_control.dm [hunk 8]
```diff
@@ -478,6 +490,47 @@ ADMIN_INTERACT_PROCS(/obj/machinery/door_control, proc/toggle)
...
+	icon = 'icons/obj/decoration.dmi'
+	icon_state = "antagscanner"
+	unpressed_icon = "antagscanner"
+	pressed_icon = "antagscanner-u"
+	unpowered_icon = "antagscanner" // should never happen, this is a failsafe if anything.
+	requires_power = 0
+	welcome_text = "Welcome, Agent."
+
+/obj/machinery/door_control/antagscanner/attack_hand(mob/user)
+	playsound(src.loc, 'sound/effects/handscan.ogg', 50, 1)
```
Comment (TobleroneSwordfish) [inline code/obj/machinery/door/door_control.dm:526]:
```suggestion
	if (ON_COOLDOWN(src, "scan", 2 SECONDS))
		return
	playsound(src.loc, 'sound/effects/handscan.ogg', 50, 1)
```
not sure what the other `ON_COOLDOWN` calls are trying to do, but this is how you'd add a cooldown to the scanner

**Review by TobleroneSwordfish — CHANGES_REQUESTED** [review]

**Comment by cheekybrdy** [comment]
!merge_upstream

**Comment by cheekybrdy** [comment]
Alarming amount of changes, but no differences on mapdiff, weird

**Pull request merged**